lib/Ohmbrewer_Thermostat.cpp

```diff
@@ -292,24 +322,24 @@ int Ohmbrewer::Thermostat::doWork() {
     //TURN ON
     if (getState() && gap!=0) {//if we want to turn on the element (thermostat is ON)
         //TURN ON state and powerPin
-        if (!(_heatingElm->getState())) {//if heating element is off
-            _heatingElm->setState(true);//turn it on
-            if (_heatingElm->getPowerPin() != -1) { // if powerPin enabled
-                digitalWrite(_heatingElm->getPowerPin(), HIGH); //turn it on (only once each time you switch state)
+        if (!(getElement()->getState())) {//if heating element is off
+            getElement()->setState(true);//turn it on
+            if (getElement()->getPowerPin() != -1) { // if powerPin enabled
+                digitalWrite(getElement()->getPowerPin(), HIGH); //turn it on (only once each time you switch state)
```

Comment:
It would be good to add a pair of methods to HeatingElement that handle flipping this pin for the modulation, if only to shorten these lines up a bit. Something like
```c++
getElement()->modulateHigh();
getElement()->modulateLow();
```
Then Thermostat wouldn't need to know which pin is the pin to modulate - it would help with encapsulation and all that.